### DISCUSSION TITLE
Can SQLPage customizations (e.g. handlebar templates) live in sqlpage_files table?

### DISCUSSION THREAD

**Discussion by shah** [sqlpage/SQLPage#174] [Q&A] [answered]
We keep all our SQLPage content in `sqlpage_files` table and it works great. Now we need to do some customizations and need to use handlebars templates but I wasn't sure if those templates could also reside in `sqlpage_files` or if those need to kept in the file system. 

Can SQLPage customizations (e.g. handlebar templates) live in sqlpage_files table?

**Comment by lovasoa** [comment, marked as answer]
Yes they can !

## Example

### Database structure

```sql
CREATE TABLE sqlpage_files(
  path VARCHAR(255) NOT NULL PRIMARY KEY,
  contents BLOB,
  last_modified TIMESTAMP DEFAULT CURRENT_TIMESTAMP
);

insert into sqlpage_files VALUES (
    'sqlpage/templates/my_component.handlebars',
    'Hello {{name}} !',
    now()
);
```

### `index.sql`:

```sql
select 'my_component' as component, 'world' as name;
```

### Result

![image](https://github.com/lovasoa/SQLpage/assets/552629/c4fb7c04-c79e-403b-b577-7a35c83b713e)

**Comment by shah** [comment]
Fantastic, thanks for a great OSS project and the quick answer!
